tests/functional/tests/test_charm.py

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+"""Functional tests for software-inventory-exporter."""
+
+import logging
+import time
+import unittest
+
+import requests
+from requests.adapters import HTTPAdapter
+from urllib3.util.retry import Retry
+from zaza import model
+from zaza.utilities.juju import get_application_ip
+
+logger = logging.getLogger(__name__)
+
+
+class SoftwareInventoryExporterTests(unittest.TestCase):
+    """Basic functional tests for software-inventory-exporter charm."""
+
+    NAME = "software-inventory-exporter"
+
+    def setUp(self) -> None:
+        """Configure resource before tests."""
+        self.session = requests.Session()
+        self.session.mount(
+            "http://", HTTPAdapter(max_retries=Retry(connect=3, backoff_factor=0.5))
+        )
+        self.endpoints = ["hostname", "kernel", "dpkg", "snap"]
+
+    def get_url(self) -> str:
+        """Get the url to be tested."""
+        port = model.get_application_config(self.NAME)["port"]["value"]
+        ubuntu_ip = get_application_ip("ubuntu")
+        return f"http://{ubuntu_ip}:{port}"
+
+    def validate_exporter(self):
```

Comment:
If it's not overly complicated, I think it'd be better to also test that the endpoint provide correct data (e.g. all snaps/kernels/packages are reported by the exporter)